### ISSUE TITLE
Have to minimize-maximize to detect controller buttons on Android build on GPD XD

### ISSUE THREAD

**Issue by bobsgame** [libretro/RetroArch#3182]
On the GPD XD running the Android nightly version of RetroArch, the controller buttons don't work until I double click the task manager button (minimize-maximize) upon which the controller is detected and everything works properly. (And it is absolutely fantastic!)
This happens every time I click Back to open the menu during gameplay, and again switching back to the gameplay itself.  It's a minor annoyance but non-intuitive to figure out the fix, so this will be helpful for others.

**Comment by fr500** [comment]
Well we don't have a GPD but it should be working fine, the guy that had one and was around to test said it was ok.

**Comment by bobsgame** [comment]
It could be a different GPD XD firmware version.  I'll update mine and see if it fixes the problem.

**Comment by bobsgame** [comment]
I updated to the newest GPD XD firmware available at this time, 3.4.6.  As with the previous firmware I was using (original 3.3.6) I have to set the game controller type to PS3 (from the default XBOX) for RetroArch to detect it at all.  

However with the latest RetroArch (2016-8-28 nightly) it has a few more problems than before, with the new firmware I still have to minimize-maximize to get controller focus, and now the A B X Y gamepad buttons automap to the Android functions Back and Search, which I cannot seem to find a way to disable.  It is impossible to play a game as is, since pressing gameplay buttons activates the menu.  It does register the button press in the game as well.  
The GPD XD has a dedicated Back button so this is unnecessary, but the firmware does seem to map those buttons to Android functions anyway, which previous versions of RetroArch may have (correctly) ignored.  I'm not sure if this is because of the new firmware or the new version of RetroArch, but it may be because of the new gamepad autoconf process mentioned in the forums.

Loading a game or a save file also takes about 30 seconds in which time RetroArch is unresponsive, whereas it was instant before.

It's probable that I have a different GPD XD hardware version than the other tester, there seem to be two different chip versions out there.

These small issues aside, RetroArch is the most amazing thing ever, and I can't be appreciative enough for these efforts.

**Comment by fr500** [comment]
Wow.. sucks that the firmware changed the keymap like that.
We haven't changed anything...

XBOX should work now too btw.

**Comment by bobsgame** [comment]
The XBOX controls are not detected at all, I've wiggled around with it quite a bit.  That is the same functionality as the previous firmware.  I don't think it's the firmware update which made the buttons remap to Android functions, I think it's how RetroArch is handling it internally.  I will try an older nightly of RetroArch from before the gamepad autoconf changes and see if it handles it correctly, if not I'll downgrade the firmware and test.

**Comment by bobsgame** [comment]
Is there an archive of older Android nightly builds anywhere?

**Comment by bobsgame** [comment]
I'll downgrade my firmware and try the latest nightly RetroArch to determine whether it is the firmware or the RetroArch build.

**Comment by bobsgame** [comment]
OK I downgraded to the oldest GPD XD firmware I could find, 3.3.6, and installed the newest nightly RetroArch, 2016-08-28.  It still has the minimize-maximize glitch to get controller focus.  You were correct, the buttons no longer map to Android functions in game, so that happened in some firmware update.

So at least with this hardware version of the GPD XD I am somewhat certain of two things:  The minimize-maximize controller focus problem, and a regression in the new build which causes a long delay when loading an old save state with Snes9X.  It also looks like Snes9X save states are no longer compatible between Snes9x cores.

It appears there are two hardware revisions of GPD XD: RK31 and RK32.  If the other tester does not have this issue, perhaps they have the other version.

**Comment by bobsgame** [comment]
I found one newer firmware version than the previous one, 3.4.7, and it seems that it does not have the buttons mapping to Android functions issue.

So the only bug here is still the minor minimize-maximize glitch, which occurs regardless of GPD XD firmware version.

**Comment by fr500** [comment]
So you start retroarch and input doesn't work till you switch apps?
To detect a controller you have to press a button on android. On my shields
I start RA, press any button and that's enough.

**Comment by bobsgame** [comment]
Correct. Every time I go from the menu to the game or the game to the menu, input is not detected until I switch apps (fine to just open the app switcher and close it) and press a button.  Pressing a button alone does not work.

**Comment by fr500** [comment]
Well a logcat could be helpful but I don't get this behavior on any of my
devices.

**Comment by bobsgame** [comment]
Here is a quick video showing the behavior.  This is a clean 3.4.7 (latest) firmware install with only the latest RetroArch nightly (2016-08-28) and no other apps installed.

https://www.youtube.com/watch?v=OOYBdl97OUA

Here is the logcat, I flipped back and forth between the game and menu like what is shown in the video.

[log.txt](https://github.com/libretro/RetroArch/files/441001/log.txt)

**Comment by bobsgame** [comment]
Small update: In the latest nightly build the problem is fixed when entering into a game, but still exists when going back to the menu.

**Comment by orbea** [comment]
@robertpelloni Is this still an issue for you?
